_inc/lib/tonesque.php

```diff
@@ -37,27 +37,7 @@ function __construct( $image_url ) {
 	}
 
 	public static function imagecreatefromurl( $image_url ) {
-	 	// Grab the extension
-		$file = strtolower( pathinfo( $image_url, PATHINFO_EXTENSION ) );
-		$file = explode( '?', $file );
-		$file = $file[ 0 ];
-
-		switch ( $file ) {
-			case 'gif' :
-				$image_obj = imagecreatefromgif( $image_url );
-				break;
-			case 'png' :
-				$image_obj = imagecreatefrompng( $image_url );
-				break;
-			case 'jpg' :
-			case 'jpeg' :
-				$image_obj = imagecreatefromjpeg( $image_url );
-				break;
-			default:
-				return false;
-		}
-
-		return $image_obj;
+		return imagecreatefromstring( file_get_contents( $image_url ) );
```

Comment:
Something I've since noticed in testing
 * Warnings will be thrown by `file_get_contents()` when the request fails
 * Warnings will be thrown by `imagecreatefromstring()` when the input is empty (`Empty string or invalid image`), or a non-image data (`Data is not in a recognized format `)

Should be noted though, that the functions being replaced would also throw PHP warnings when invalid URLs or images were passed, so nothing has changed there.